### ISSUE TITLE
[Bug]: if an app exists in `apps` and `custom_apps` folder, `apps` will be picked first for js and css ressources 

### ISSUE THREAD

**Issue by skjnldsv** [nextcloud/server#37043]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When you use `addScript` and request a script of your app to be loaded
AND you have different versions of the app in both of your apps folders
THEN you will have ressources loaded from the apps folder, even if the version is lower

### Steps to reproduce

1. Setup a `custom_apps` folder
2. Install latest talk in `custom_apps`
3. Install old talk (11.x.x) in `apps` folder
4. Load the page
  ![image](https://user-images.githubusercontent.com/14975046/222788283-7a5f4ce4-5c0b-463f-bc82-9603231d97d2.png)

### Expected behavior

All the ressources requests should be loaded from the default apps root instead

### Installation method

Community Manual installation with Archive

### Operating system

Other

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
-
```

**Comment by szaimen** [comment]
cc @ArtificialOwl as this looks like a pretty big problem with multiple app versions.

**Comment by joshtrichards** [comment]
Is there a good reason for supporting multiple versions of an app being installed simultaneously? Only one can truly be active at a time anyhow. 

I think we should just add an admin setup check that considers this an invalid situation and tells the admin to eliminate the duplicate apps (i.e. #40063).

**Comment by szaimen** [comment]
Yes, the idea was to allow to update shipped app via the appstore...

**Comment by szaimen** [comment]
@ArtificialOwl now that you released https://github.com/nextcloud-releases/related_resources/releases/tag/1.2.0 to the store, https://apps.nextcloud.com/apps/related_resources, I suppose people will run into the above issue. So would you mind having a look at the problem above?

**Comment by joshtrichards** [comment]
> Yes, the idea was to allow to update shipped app via the appstore...

Maybe we make the logic prefer the `writable` apps path (under the assumption it's the one most likely to be the newest/updated/from-the-app-store one).
